resources/assets/coffee/react/beatmapset-page/main.coffee

```diff
@@ -49,36 +49,50 @@ BeatmapsetPage.Main = React.createClass
           currentPlaymode = mode
           break
 
+    @state =
+      beatmaps: beatmaps
+      beatmapList: beatmapList
+      currentBeatmapId: currentBeatmapId
+      currentPlaymode: currentPlaymode
+      loading: false
+      isPreviewPlaying: false
+      currentScoreboardType: 'global'
+      enabledMods: []
+      scores: []
+      userScore: null
+      userScorePosition: -1
 
-    beatmaps: beatmaps
-    beatmapList: beatmapList
-    currentBeatmapId: currentBeatmapId
-    currentPlaymode: currentPlaymode
-    loading: false
-    isPreviewPlaying: false
-    currentScoreboard: 'global'
-    scores: []
-
-  setHash: ->
+  setHash: =>
     osu.setHash BeatmapsetPageHash.generate
-      page: @state.currentPage
       beatmapId: @state.currentBeatmapId
       playmode: @state.currentPlaymode
 
-  setCurrentScoreboard: (_e, {scoreboard, forceReload = false}) ->
+  setCurrentScoreboard: (_e, {scoreboardType = @state.currentScoreboardType, enabledMod = null, forceReload = false, resetMods = false}) =>
     return if @state.loading
```

Comment:
i mean right now this is supposed not to let you start a new download if there is another in progress, so that something like this shouldn't actually happen